rewrite-java-core/src/test/java/CredentialTest.java

```diff
@@ -0,0 +1,48 @@
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.ChangePackage;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+/**
+ * CredentialTest is used to test out the recipe that changes the package name com.azure.core.credential
+ * to io.clientcore.core.credential.
+ * @author Ali Soltanian Fard Jahromi
+ */
+public class CredentialTest implements RewriteTest {
+
+    /**
+     * This method sets which recipe should be used for testing
+     * @param spec stores settings for testing environment; e.g. which recipes to use for testing
+     */
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResource("/META-INF/rewrite/rewrite.yml",
+                "com.azure.rewrite.java.core.MigrateAzureCoreSamplesToAzureCoreV2");
+    }
+
+    /**
+     * This test method is used to make sure that the package com.azure.core.credential is changed
+     */
+    @Test
+    void testCredentialPackageNameChange() {
+        @Language("java") String before = "import com.azure.core.credential.KeyCredential;";
+        before += "\npublic class Testing {";
+        before += "\n  public Testing(){";
+        before += "\n    com.azure.core.credential.KeyCredential kc = new KeyCredential(\"<api-key>\");";
```

Comment:
Same here - we should not use fully qualified names when there's an import. Also, we should add another test where there's not import and the fully qualified name is used.

Reply:
Done in commit [492daeb](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/492daeb48360470829cc54218911b9b3e1758ace)